director/scripts/test/kinematic_test.py

```diff
@@ -56,19 +75,84 @@ def state_handler(self, channel, data):
         self.plant.SetPositions(self.context, msg.position)
         self.plant.SetVelocities(self.context, msg.velocity)
 
-        # TODO: pt_body and body_name are JSON options
-        # Once you can visualize, play around with these values to see what
-        # they do. Look in the .urdf file above to know valid body names
-        pt_body = np.array([0,0,0])
-        body_name = "toe_left"
-
-        # Use Drake's CalcPointsPositions to determine where that point is
-        # in the world
-        pt_world = self.plant.CalcPointsPositions(self.context,
-            self.plant.GetFrameByName(body_name), pt_body,
-            self.plant.world_frame())
-        print(pt_world.transpose())
-        #TODO instead of printing the position, visualize it
+        for data in self.data['data']:
+            jsonData = eval(str(data))
+
+            # set the body and point on which to visualize the data
+            pt_body = np.array(jsonData['point-on-body'])
+            body_name = jsonData['body_part']
+
+            # Use Drake's CalcPointsPositions to determine where that point is
+            # in the world
+            pt_world = self.plant.CalcPointsPositions(self.context,
+                self.plant.GetFrameByName(body_name), pt_body,
+                self.plant.world_frame())
+            next_loc = pt_world.transpose()[0]
+
+            # draw a continuous line
+            if (jsonData['type'] == "line"):
+
+                # check if this line has bee drawn or not
+                try:
```

Comment:
Never expect failure from a try-catch. It's very computationally expensive. The way you do this for the point is better.